### PR TITLE
Require login

### PR DESCRIPTION
**Pull request by preciousonah**
**Description**

- Add functionality that prevents unauthenticated users from accessing pages on the website

**Milestones**

Wrap all the components with RequireLogin function which prevents users that are not logged in from accessing my website
Here's a preview of my website: 
Image 8-4-23 at 1 26 AM
- I'm working on the UI for all the pages in my website so i will improve in the styling of the login page soon!

### CODE REVIEW

File: fashion-messaging-app-ui/src/RequireLogin.jsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Navigate, useLocation } from 'react-router-dom';
```
Comment (preciousonah) [inline fashion-messaging-app-ui/src/RequireLogin.jsx:1]: Navigate helps me change current routes, useLocation checks what location/route my app is in

File: fashion-messaging-app-ui/src/RequireLogin.jsx [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import { useContext } from 'react';
+import { UserContext } from './UserContext';
+
+export default function RequireLogin({ children }) {
+  const { user } = useContext(UserContext);
```
Comment (preciousonah) [inline fashion-messaging-app-ui/src/RequireLogin.jsx:6]: The useContext helps me get the logged-in user information

File: fashion-messaging-app-ui/src/RequireLogin.jsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  const location = useLocation();
+ 
+  if (!user) {
+    return <Navigate to="/login" state={{ from: location }} replace />
+}
```
Comment (preciousonah) [inline fashion-messaging-app-ui/src/RequireLogin.jsx:11]: This means if the user is not logged in, it should route to the login route and the user cannot be redirected to anywhere in the website until they are authenticated

File: fashion-messaging-app-ui/src/App.jsx [hunk 4]
```diff
@@ -8,11 +8,14 @@ import LoginForm from './components/LoginForm/LoginForm';
...
 import Recommendations from './components/Recommendations/Recommendations';
+import RequireLogin from './RequireLogin';
+
 
 export default function App() {
   const [loading, setLoading] = useState(true);
   const [user, setUser] = useState(() => {
-    return null;
+    const storedUser = localStorage.getItem('user');
+    return storedUser ? JSON.parse(storedUser) : null;
```
Comment (preciousonah) [inline fashion-messaging-app-ui/src/App.jsx:18]: This gets the user information that was previously stored in the localStorage

File: fashion-messaging-app-ui/src/App.jsx [hunk 5]
```diff
@@ -53,6 +56,11 @@ export default function App() {
     setSearchQuery(query);
   };
 
+
+  useEffect(() => {
+    localStorage.setItem('user', JSON.stringify(user));
+  }, [user]);
```
Comment (preciousonah) [inline fashion-messaging-app-ui/src/App.jsx:62]: This stores the user information in local storage

File: fashion-messaging-app-ui/src/RequireLogin.jsx (hunk 3 above)
Comment (speltzka) [reply fashion-messaging-app-ui/src/RequireLogin.jsx:11]: Nice! I like this change

File: fashion-messaging-app-ui/src/App.jsx (hunk 5 above)
Comment (speltzka) [reply fashion-messaging-app-ui/src/App.jsx:62]: Do you need to store the user in local storage if you can get the user from useContext(UserContext) ?

**Review by speltzka — COMMENTED** [review]
Reviewed! I don't totally understand why we need to use localstorage, let me know! :)

**Pull request merged**